src/main/scala/cognite/spark/v1/RelationshipsRelation.scala

```diff
@@ -0,0 +1,156 @@
+    val filtersAsMaps = pushdownToParameters(pushdownFilterExpression)
+
+    val relationshipsFilterSeq = if (filtersAsMaps.isEmpty || shouldGetAllRows) {
+      Seq(RelationshipsFilter())
+    } else {
+      filtersAsMaps.distinct.map(relationshipsFilterFromMap)
+    }
+
+    val streamsPerFilter = relationshipsFilterSeq
+      .map { f =>
+        client.relationships.filterPartitions(f, numPartitions, limit)
+      }
+
+    // Merge streams related to each partition to make sure duplicate values are read into
+    // the same RDD partition
+    streamsPerFilter.transpose
+      .map(s => s.reduce(_.merge(_)))
+  }
+
+  def relationshipsFilterFromMap(m: Map[String, String]): RelationshipsFilter =
+    RelationshipsFilter(
+      sourceExternalIds = m.get("sourceExternalIds").map(stringSeqFromWrappedArray),
+      sourceTypes = m.get("sourceTypes").map(stringSeqFromWrappedArray),
+      targetExternalIds = m.get("targetExternalIds").map(stringSeqFromWrappedArray),
+      targetTypes = m.get("targetTypes").map(stringSeqFromWrappedArray),
+      dataSetIds = m.get("dataSetId").map(idsFromWrappedArray(_).map(CogniteInternalId)),
+      startTime = timeRangeFromMinAndMax(m.get("minStartTime"), m.get("maxStartTime")),
+      endTime = timeRangeFromMinAndMax(m.get("minEndTime"), m.get("maxEndTime")),
+      activeAtTime = timeRangeFromMinAndMax(m.get("minActiveAtTime"), m.get("maxActiveAtTime")),
+      labels = getLabelsFilter(m.get("labels")),
+      confidence = confidenceRangeFromMinAndMax(m.get("minConfidence"), m.get("maxConfidence")),
+      createdTime = timeRangeFromMinAndMax(m.get("minCreatedTime"), m.get("maxCreatedTime")),
+      lastUpdatedTime = timeRangeFromMinAndMax(m.get("minLastUpdatedTime"), m.get("maxLastUpdatedTime"))
+    )
+
+  override def insert(rows: Seq[Row]): IO[Unit] = {
+    val relationships = rows.map(fromRow[RelationshipCreate](_))
+    client.relationships
+      .create(relationships)
+      .flatTap(_ => incMetrics(itemsCreated, relationships.length)) *> IO.unit
+  }
+
+  override def delete(rows: Seq[Row]): IO[Unit] = {
+    val relationshipIds = rows.map(fromRow[RelationshipsDeleteSchema](_)).map(_.externalId)
+    client.relationships
+      .deleteByExternalIds(relationshipIds)
+      .flatTap(_ => incMetrics(itemsDeleted, relationshipIds.length))
+  }
+
+  override def upsert(rows: Seq[Row]): IO[Unit] =
+    throw new CdfSparkException("Upsert is not supported for relationships")
+
+  override def update(rows: Seq[Row]): IO[Unit] =
+    throw new CdfSparkException("Update is not supported for relationships")
+}
+
+object RelationshipsRelation {
+  var insertSchema: StructType = structType[RelationshipsInsertSchema]
+  var readSchema: StructType = structType[RelationshipsReadSchema]
+  var deleteSchema: StructType = structType[RelationshipsDeleteSchema]
+}
+
+final case class RelationshipsDeleteSchema(
+    externalId: String
+)
+
+final case class RelationshipsInsertSchema(
```

Comment:
@hakontro I saw these classes are defined in all data types but I could not catch where they are being used looking at SdkV1Relation. Could you point me there?

For labels, it is defined as "labels: Option[Seq[Map[String, String]]]" considering it can be translated into Option[Seq[CogniteExternalId]]. Is it a correct approach?

Reply:
These are actually used in the update and upsert methods, we don't need them yet since we aren't implementing those :) But just leave them there, we'll do that soon!

Reply:
RIght click class name -> Find Usages will often help you find stuff like this!

Reply:
> For labels, it is defined as "labels: Option[Seq[Map[String, String]]]" considering it can be translated into Option[Seq[CogniteExternalId]]. Is it a correct approach?
Not sure where you're referring to here, can you point me to it?

Reply:
I think this question was nonsense since I did not understand this was for update/upsert.